### ISSUE TITLE
Add grep support for cat APIs

### ISSUE THREAD

**Issue by ebuildy** [elastic/elasticsearch#30245]
We ``grep`` often the output of ``_cat`` API to find what we are looking for, why not add a parameter to do this directly?

``curl http://localhost:9200/_cat/indices?grep=events-``

This will make possible:
* Usage on a browser
* Share relevant URL with the grep info

**Comment by tvernum** [comment]
What exactly are you grepping for?
If that's just an index name (which seems likely given that the output of `_cat/indices` doesn't contain much in the way of grep'able text) then you can just include the index patten in the URL:
Per: https://www.elastic.co/guide/en/elasticsearch/reference/6.2/cat-indices.html#cat-indices
You can do 
```
curl http://localhost:9200/_cat/indices/events-*
```
(or `*events-*` depending on what you're trying to match)

**Comment by ebuildy** [comment]
Ho it was a bad example ^^

What about finding all indices that are in yellow state? or using the _cat segments API.

**Comment by mP1** [comment]
@ebuildy 

That very first example is given on the link...

> Examplesedit
> Which indices are yellow?
> 
> GET /_cat/indices?v&health=yellow

**Comment by jasontedor** [comment]
I do not think that we should add this functionality. The cat APIs are by design meant for the command line where tools like `grep` are readily available. I do not want us to carry the weight of maintaining features like this nor that of complicating our APIs.
